src/gui/editors/AutomationEditor.cpp

```diff
@@ -492,162 +400,228 @@ void AutomationEditor::drawLine( int x0In, float y0, int x1In, float y1 )
 
 void AutomationEditor::mousePressEvent( QMouseEvent* mouseEvent )
 {
-	QMutexLocker m( &m_patternMutex );
 	if( !validPattern() )
 	{
 		return;
 	}
-	if( mouseEvent->y() > TOP_MARGIN )
+
+	// If we clicked inside the AutomationEditor viewport (where the nodes are represented)
+	if (mouseEvent->y() > TOP_MARGIN && mouseEvent->x() >= VALUES_WIDTH)
 	{
 		float level = getLevel( mouseEvent->y() );
 
-		int x = mouseEvent->x();
+		// Get the viewport X
+		int x = mouseEvent->x() - VALUES_WIDTH;
 
-		if( x >= VALUES_WIDTH )
-		{
-			// set or move value
+		// Get tick in which the user clicked
+		int posTicks = (x * TimePos::ticksPerBar() / m_ppb) + m_currentPosition;
 
-			x -= VALUES_WIDTH;
+		// Get the time map of current pattern
+		timeMap & tm = m_pattern->getTimeMap();
 
-			// get tick in which the user clicked
-			int pos_ticks = x * TimePos::ticksPerBar() / m_ppb +
-							m_currentPosition;
+		m_mouseDownLeft = (mouseEvent->button() == Qt::LeftButton);
+		m_mouseDownRight = (mouseEvent->button() == Qt::RightButton);
 
-			// get time map of current pattern
-			timeMap & time_map = m_pattern->getTimeMap();
+		// Some actions require that we know if we clicked the inValue of
+		// a node, while others require that we know if we clicked the outValue
+		// of a node.
+		bool editingOutValue = (
+			m_editMode == DRAW_OUTVALUES
+			|| (m_editMode == ERASE && m_mouseDownRight)
+		);
 
-			// will be our iterator in the following loop
-			timeMap::iterator it = time_map.begin();
+		timeMap::iterator clickedNode = getNodeAt(mouseEvent->x(), mouseEvent->y(), editingOutValue);
 
-			// loop through whole time-map...
-			while( it != time_map.end() )
+		switch(m_editMode)
+		{
+			case DRAW:
```

Comment:
We don't have a rule for indenting switch/case statements and thus have inconsistent indentations in many files. It should probably be discussed later.
BTW, you should add a space between `switch` and `(`.

Reply:
I have no preference on whether we indent `case` or not. I typically do, but if we want to save indentations then we can keep it aligned with `switch.